### PR TITLE
Workflow and pre-commit added

### PR DESCRIPTION
**Pull request by uk-bolly**
**Overall Review of Changes:**

- new workflow configuration
  - Allowing devel and main configs
  - IaC code found in alternate repo for easier mgmt
  - separate pipeline configs
  - new pre-commit testing
  - linting tested
- Added pre-commit config - Does not have to be used but can improve things
  - .pre-commit-config.yaml
  - .config/.secrets.baseline

- updated to logic in 5.6.5
- lint updates to 6.1.x
- readme updates
- audit control updates and variable name changes
  - ability to run audit on arm64(e.g. pi or M1/2) too thanks to @lucab85 #77
- tidy up README adopted PR #78 thanks to @lucab85
- moved Makefile requirements.txt to .config/
- removed .ansible.cfg and local.yml
- README button updates

**How has this been tested?:**
separate pipeline testing and repo
manually

### CODE REVIEW

**Comment by uk-bolly** [comment]
expect pre-commit to fail as not yet in the commited repo and
expect current pipeline to fail and files no longer exist in repo

File: README.md
Comment (MrSteve81) [inline README.md:1]: I know it minor but to keep the readme looking pretty your github tag needs to be done like this for the pieplines. 
[![Main Pipeline Status]
[![Devel Pipeline Status]

**Review by MrSteve81 — CHANGES_REQUESTED** [review]
I know its minor but to keep the readme looking pretty your github tag needs to be done like this for the pieplines.
[![Main Pipeline Status]
[![Devel Pipeline Status]

**Review by georgenalen — APPROVED** [review]
Looks good!

**Pull request merged**